numpy/polynomial/polynomial.py

```diff
@@ -780,6 +781,95 @@ def polyval(x, c, tensor=True):
     return c0
 
 
+def polyvalfromroots(x, r, tensor=True):
+    """
+    Evaluate a polynomial specified by its roots at points x.
+
+    If `r` is of length `N`, this function returns the value
+
+    .. math:: p(x) = \prod_{n=1}^{N} (x - r_n)
+
+    The parameter `x` is converted to an array only if it is a tuple or a
+    list, otherwise it is treated as a scalar. In either case, either `x`
+    or its elements must support multiplication and addition both with
+    themselves and with the elements of `r`.
+
+    If `r` is a 1-D array, then `p(x)` will have the same shape as `x`.  If
+    `r` is multidimensional, then the shape of the result depends on the
+    value of `tensor`. If `tensor` is true the shape will be r.shape[1:] +
+    x.shape. If `tensor` is false the shape will be r.shape[1:]. Note that
+    scalars have shape (,).
+
+    Parameters
+    ----------
+    x : array_like, compatible object
+        If `x` is a list or tuple, it is converted to an ndarray, otherwise
+        it is left unchanged and treated as a scalar. In either case, `x`
+        or its elements must support addition and multiplication with
+        with themselves and with the elements of `r`.
+    r : array_like
+        Array of roots. If `r` is multidimensional the remaining indices
+        enumerate multiple polynomials. In the two dimensional case the
+        coefficients may be thought of as stored in the columns of `r`.
+    tensor : boolean, optional
+        If True, the shape of the coefficient array is extended with ones
+        on the right, one for each dimension of `x`. Scalars have dimension 0
+        for this action. The result is that every column of coefficients in
+        `r` is evaluated for every element of `x`. If False, `x` is broadcast
+        over the columns of `r` for the evaluation.  This keyword is useful
+        when `r` is multidimensional. The default value is True.
+
+        .. versionadded:: 1.12
```

Comment:
This should be after the summary, it is here for polyval because it was added in a later release.